### ISSUE TITLE
Problems during initial startup of VS2019 after installation

### ISSUE THREAD

**Issue by gh-Destiny** [Nesviatypaskha/VsConsoleOutput#5]
Hi! Your fabulous tool was installed from official VS market. W/O problems. Latest ver. 1.1.2
But! After the reboot and initial start of VS 2019 (16.11.42) I just got the error message at the dialog
which said that I must take a look in the file C:\Users\user_name\AppData\Roaming\Microsoft\VisualStudio\16.0_f06fdb9c\ActivityLog.xml
And inside it - at the end I could read that:

```
    <description>CreateInstance failed for package [VSConsoleOutputPackage]Source: 'mscorlib' Description: Could not load file or assembly 'Microsoft.VisualStudio.Shell.15.0, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. 
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.VisualStudio.Shell.15.0, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. 
File name: 'Microsoft.VisualStudio.Shell.15.0, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
	at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type)
	at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)
	at System.Activator.CreateInstanceFromInternal(String assemblyFile, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo)
	at System.AppDomain.CreateInstanceFrom(String assemblyFile, String typeName) 

WRN: Assembly binding logging is turned OFF. 
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1. 
Note: There is some performance penalty associated with assembly bind failure logging. 
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog]</description>
```

The closest and practically the only analogue of this error I could find here
https://github.com/GoogleCloudPlatform/google-cloud-visualstudio/issues/1148
But the solution there, of course, turned out to be not very accessible - it was suggested to simply install the previous release.
BUT! your tool does not have a version archive anywhere. so there was no way I could carry out this proposal.

I hope for a very quick solution to this problem, because the absence of this necessary thing in VS2019 is 
like viewing the console output of all internal debugging commands greatly reduces my overall performance.
I switched to studio from QtCreator - and there the console was - immediately built in and I saw all the internal outputs
functions for debugging information. Such as qDebug, qInfo, qWarning. And now I don’t see anything like this anywhere...

**Comment by gh-Destiny** [comment]
I tried also to find on my system "Microsoft.VisualStudio.Shell.15.0, Version=17.0.0.0" - but failed.
The only version which I have preinstalled is 16.0.0.0! and not 17.0.0.0! 
So from where does this request for loading 17 come from?

**Comment by Nesviatypaskha** [comment]
HI, I'm on medical live, please ping me next week, and I will be able to
support you. maybe build custom version for you

пт, 29 нояб. 2024 г. в 11:17, gh-Destiny ***@***.***>:

**Comment by gh-Destiny** [comment]
All the benefits to you and health and professional medical staff)

**Comment by gh-Destiny** [comment]
I'm not sure it's time for a polite reminder - but you suggested it yourself ;).
So I ask you to return to this problem, if possible.

**Comment by Nesviatypaskha** [comment]
Hi!
Today will work with your problem, thanks for your reminder)

пт, 13 груд. 2024 р. о 08:30 gh-Destiny ***@***.***> пише:

**Comment by gh-Destiny** [comment]
https://stackoverflow.com/questions/46054343/visual-studio-2015-fail-to-load-package-because-microsoft-visualstudio-shell-15
If this article is to be believed, the problem is that the level of versions for which this plugin is created is raised too high.
I.e. It feels like you've only made it actually available for the VS2022 version and their future variants.

Very probably but it should be available from VS2015 at least ;)

For my current VS2019 Microsoft.VisualStudio.Shell.15.0.dll has 16.0.30712.41 ver. dated by 28/07/2022
in path ...\2019\Enterprise\Common7\IDE\Extensions\2qyepqqu.524
Ver. of VS2019 - is 16.11.42
.NET Framework - is 4.8.09037

**Comment by Nesviatypaskha** [comment]
Yes, I know the root cause but I couldn't figure out why it failed on your
side. VS2019 and VS2022 have two different APIs that it need to use for
extension. My next step will be to setup separate windows and your version
on Visual Studio. On my PC installed ~100gb of SDKs for visual code and I
think need clear setup for reproduce

пн, 16 дек. 2024 г. в 10:22, gh-Destiny ***@***.***>:

**Comment by Nesviatypaskha** [comment]
Could you please try building locally vsix on your PC? I will able to
continue on Friday

вт, 17 дек. 2024 г. в 22:56, Алексей Несвятипасха ***@***.***>:

**Comment by gh-Destiny** [comment]
> building locally vsix

You mean - to compile from github sources with help of [VsConsoleOutput.sln] solution file
in my VS2019 your tool? I already tried and got massive block of non-understandable errors.
So that's why I decided to open an issue here and not to try to resolve this issue locally...

**Comment by Nesviatypaskha** [comment]
Yes. You just need to install SDK for the extension development. I tried
the last version of VS19, the extension work fine

чт, 19 дек. 2024 г. в 10:33, gh-Destiny ***@***.***>:

**Comment by gh-Destiny** [comment]
Definitely not just download and install the SDK - but something else definitely needs to be done.
Because just when opening the solution - two of internal sub-projects were fixed in the "unloaded" state 
and I can't load them. This clearly should not happen.
And it's also unclear - which of the projects is the main one and who is creating the extension file that needs to be put where? 
You need to provide me with a complete instruction on how to assemble and how to install the result of the assembly.

P.S. I definitely do not understand which exactly SDK it requires. I thought I have 'em all (6 & 8 Long-term editions). But console.proxy.cs project is still with unloaded state and gives the error "The project file cannot be opened. Unable to locate the .NET SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version". But which precisely version is it??? who knows? And I do not have global.json file.

Also it's strange that you project is requesting PlatformToolset v143 but for VS2019 v142 is more regular and default...
You definitely DO NOT support the VS workflow with **minimum specs**.

I don't understand how to correctly build your solution in such conditions...

According to this site: https://dotnet.microsoft.com/en-us/download/visual-studio-sdks
I can only to download and install SDK for version [.NET 5.0]! from a link https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/sdk-5.0.408-windows-x64-installer  -  If I use VS2019...
But I already have runtimes for versions 6 and 8. Moreover, I’m thinking of deleting 6 already. And install 9th.

**Comment by Nesviatypaskha** [comment]
The main project is an extension. need to build all 3 projects.  C++ is
additional and just added in extention

чт, 26 дек. 2024 г. в 11:10, gh-Destiny ***@***.***>:

**Comment by gh-Destiny** [comment]
The main question remains unanswered ;) if you did not catch it ))

"You definitely DO NOT support the VS workflow with **minimum** specs."

I can't do anything with VS2019 because you have set the solution to work with VS2022+ ONLY
AND I still do not understand which VERSION exactly of which SDK exactly I must install in addition
to all of my existed ones: give me **a full URL** to it please. We only MUST take into account that all this 
should work on VS2019! So pls, introduce back all changes which we must made for solution/projects 
in order to be able to open and load all of them on VS2019!

In other words, all files in the solution should be minimally opened and built under VS2019.
And the statement of support for VS2022 should only mean that when trying to open such a solution,
with this VS - it will just offer to make an upgrade for PlatformToolset - but after that it will still be quietly 
compiled....

**Comment by Nesviatypaskha** [comment]
Seriously? )))

Visual Studio changed internal SDK version specifications for 2019 and
2022. I had great sex with this and do not know what happened inside MS
I tried it with my student in 2019, and it works!

Microsoft.VisualStudio.Shell.15.0 that mantioned is not 2019.

After migration form 2019 to 2022 MS broke compatibility and it was
need to update manifest and nugets from SDK 2019v1 to SDK2019v2.


You could use previous version, please downgrade git version, try to
build on your enviroment.

https://learn.microsoft.com/en-us/visualstudio/extensibility/visual-studio-sdk?view=vs-2022


What you need to do it is do first steps from instruction and instead
of create new extention clone repository.

Build one by one c++ projects and extention.

You will see in folder two libs and extention staff. It packed also in
vsix file.


Carently we have holidays. I two years not work on win pc and I'm not
able to come in university to use widows PC. I tried to run VS on
virtual machine, but it is omg on mac.


In any case, I will not leave you with your problem and will try to
solve it myself or help you solve it locally


пт, 27 дек. 2024 г. в 15:57, gh-Destiny ***@***.***>:

**Comment by Nesviatypaskha** [comment]
This is the old version of src that was used in 2019. Please try it. All
new version was built in 2022

вт, 31 дек. 2024 г. в 01:19, Алексей Несвятипасха ***@***.***>:

**Comment by Nesviatypaskha** [comment]
https://drive.google.com/file/d/1SbEEt2DYYWKcmJO0MfWqttc8xGMG8v4w/view?usp=sharing
This works on 2019 on PC of my student. Try it
[image: image.png]

вт, 31 дек. 2024 г. в 01:42, Алексей Несвятипасха ***@***.***>:

**Comment by Nesviatypaskha** [comment]
In any case, be very careful with MS, versions, .net and so on.

вт, 31 дек. 2024 г. в 02:14, Алексей Несвятипасха ***@***.***>:

**Comment by gh-Destiny** [comment]
Of course, first accept my congratulations on the New Year and Christmas))
These family holidays deserve to be spent with families))
In a warm circle of relatives and friends! All the best to you!

As for this project... Both MS and its "crafts" are starting to me off.
In the process of trying to install your old version, I somehow managed to have a state when a plugin does exist,
BUT there is no way to understand where it is and how to reinstall/remove it. 
All the necessary action menu items simply do not work or are hidden. Just an incredible coincidence

But even after - as I was able to install the plugin - I don't see it working.
That is, like in your picture <a href="https://ibb.co/rpHvpjL"><img src="https://i.ibb.co/8x9XxCR/preview-cpp-1.png" alt="preview-cpp-1" border="0"></a> - a new menu item 'Console' for Output panel has appeared in mine VS2019 env, 
but nothing is ever withdrawn into it. ALTHOUGH I know 100% that there is an output printf's in principle.

As for attempts to compile the entire solution, I ask you to clarify once again which .NET SDKs
are required - up to their full URL-path on the Internet - in order to download and install them.
Just by itself - nothing is uploaded or compiled here.

**Comment by Nesviatypaskha** [comment]
https://developercommunity.visualstudio.com/t/Installation-of-wixtoolset-extension-cor/10057006
https://stackoverflow.com/questions/22444799/how-to-uninstall-vsix-visual-studio-extensions
You could remove in manually and try to reinsatll

вт, 31 дек. 2024 г. в 13:12, gh-Destiny ***@***.***>:

**Comment by gh-Destiny** [comment]
Thx, it will be useful for the next stages, but right now, I was able to perform this operation alone.
SO what is left - I already wrote about it: I do not see the working state of this "old" plugin.
And as for attempts to compile the entire solution, I ask you to clarify once again which .NET SDKs
are required - up to their full URL-path on the Internet - in order to download and install them.
Just by itself - nothing is uploaded or compiled here.

**Comment by Nesviatypaskha** [comment]
Just build git
[image: image.png]
[image: image.png]

[image: image.png]
[image: image.png]
[image: image.png]

вт, 31 дек. 2024 г. в 22:15, gh-Destiny ***@***.***>:

**Comment by gh-Destiny** [comment]
If you wanted to show something via some pics - this attempt failed:
We see only
```
[image: image.png]
[image: image.png]

[image: image.png]
[image: image.png]
[image: image.png]
```

And I still need a full URL-path to .NET SDK!
By itself this software product will not appear on my computer.
And therefore nothing can be successfully compiled.

**Comment by Nesviatypaskha** [comment]
may be you have any messenger?

ср, 1 янв. 2025 г. в 21:44, gh-Destiny ***@***.***>:
